pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -118,6 +122,29 @@ func convertService(svc v1.Service, domainSuffix string) *model.Service {
 	}
 }
 
+func externalNameServiceInstances(k8sSvc v1.Service, svc *model.Service) []*model.ServiceInstance {
+	if k8sSvc.Spec.Type != v1.ServiceTypeExternalName || k8sSvc.Spec.ExternalName == "" {
+		return nil
+	}
+	var out []*model.ServiceInstance
+	for _, portEntry := range svc.Ports {
+		out = append(out, serviceInstance(k8sSvc, portEntry, svc))
+	}
+	return out
+}
+
+func serviceInstance(k8sSvc v1.Service, portEntry *model.Port, svc *model.Service) *model.ServiceInstance {
```

Comment:
I would inline this function in case some others misuse it.

Reply:
Done